### ISSUE TITLE
Evaluate impacts of UX on UI

### ISSUE THREAD

**Issue by paoloantinori** [syndesisio/syndesis#2645]

**Comment by amysueg** [comment]
Tagging @seanforyou23 and @michael-coker for investigation. FYI @gashcrumb @sjcox-rh @dongniwang.

**Comment by gashcrumb** [comment]
Might as well just assign this @amysueg :-)

**Comment by amysueg** [comment]
Wasn't sure of the protocol but now I know! Thx @gashcrumb

**Comment by gashcrumb** [comment]
@heiko-braun think it's safe to say this isn't gonna happen in sprint 31, but this is a post 7.1 effort anyways, going to proactively move it to the next sprint.

**Comment by heiko-braun** [comment]
Thanks @gashcrumb

**Comment by michael-coker** [comment]
Per [the design](https://redhat.invisionapp.com/share/X6K0SKUWZFG#/screens/300744577)...

Some changes I could make:

* Create “steps” card view layout, including empty state card. Should take a couple of days.
* Create mini card view for step and connection cards, and toggle button that switches between normal/mini card view. This would likely take a couple of days as well.
* Update IVP link to “add to integration” that links to a new page (mentioned below) that contains tabs with card views for steps, connections, and steps+connections. This, and the rest of the bullets in this list should also take a couple of days.
* Update IVP wording while in the process of adding a step/connection
* Add “create extension” button/link to empty state card in “all” tab
* Add tooltip to empty state card on “all” tab that explains difference between connection and extension.
* Add info icon and popover on “connection” tab that explains what connections are.
* Update empty state page copy and "add to integration" button.

I’m not sure about the details or how long the following tasks will take. It’s more angular work than I’m familiar with:

* Create the new page that has tabs for “all”, “connections”, and “steps”, with “all” selected by default
 * This involves creating new views for “all” and “steps” that display when you choose the corresponding tab - I’m assuming we can use the existing “connections” view.
* Add tabs to the new page that display the individual all, connections, and steps views.
* Add a filter toolbar to each tab that filters cards in that tab by connection/step name.

**Comment by gashcrumb** [comment]
Yeah, so I'd probably take out the current 'connection-select' and 'step-select' pages and replace then with a new page that fetches/combines the steps and connections into one array.  Then it's a matter of ensuring for the start/finish positions that we filter out steps, assuming that integrations still only start/end with a connection, and ensuring that the code properly branches to either the 'action-select' or 'step-configure' pages based on what type the user selects, none of this seems too bad.

The tabs then would just filter the combined list by type on the new page, and I guess in that way we could also ensure that the page filters to connections only as needed depending on the position in the flow.

And then yeah, we've a lot of 'add a step/add a connection' links to take out which will be awesome actually :-).
